### ISSUE TITLE
`__version__` string of conda-forge `cuda-python` 12.6.2 & 11.8.5 build 1 is dirty

### ISSUE THREAD

**Issue by leofang** [NVIDIA/cuda-python#308]
```
+ python -c 'import cuda.bindings; print(cuda.bindings.__version__)'
0+untagged.310.ge14f0f1.dirty
```
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1107951&view=logs&j=5b44946f-b0b5-570f-e735-950feabb4722&t=61cfdd87-b98a-5bb9-08e2-446c3dd957d8&l=2050

This affects `cuda.core` logic, ex:
https://github.com/NVIDIA/cuda-python/blob/eeda1ab5b3b89b1e8ee8ae8dc2f1cbff502b2d5c/cuda_core/cuda/core/experimental/_module.py#L33

**Comment by leofang** [comment]
12.6.2 is fixed (https://github.com/conda-forge/cuda-python-feedstock/pull/100). Need to fix 11.8.5 (https://github.com/conda-forge/cuda-python-feedstock/issues/104).

**Comment by leofang** [comment]
Both 12.6.2 & 11.8.5 build 2 are out. Root cause is that the git archived generated by GitHub did not have the git keywords properly populated (https://github.com/conda-forge/cuda-python-feedstock/pull/100#issuecomment-2558790174), most likely due to some versioneer's quirky behaviors
